modules/gapi/test/common/gapi_video_tests_common.hpp

```diff
@@ -0,0 +1,239 @@
+                                                    params.flags, params.minEigThreshold);
+
+        cv::GComputation c(cv::GIn(inPrev, inNext, prevPts, predPts),
+                           cv::GOut(nextPts, statuses, errors));
+
+        c.apply(cv::gin(in.prevData, in.nextData, in.prevPoints, std::vector<cv::Point2f>{ }),
+                cv::gout(gapiOut.nextPoints, gapiOut.statuses, gapiOut.errors),
+                std::move(const_cast<cv::GCompileArgs&>(params.compileArgs)));
+
+        return c;
+    }
+}
+
+inline cv::GComputation runOCVnGAPIOptFlowLK(TestFunctional& testInst,
+                                             std::vector<cv::Point2f>& inPts,
+                                             const OptFlowLKTestParams& params,
+                                             OptFlowLKTestOutput& ocvOut,
+                                             OptFlowLKTestOutput& gapiOut)
+{
+    testInst.initMatsFromImages(params.channels,
+                                params.fileNamePattern,
+                                params.format);
+
+    OptFlowLKTestInput<cv::Mat> input{ testInst.in_mat1, testInst.in_mat2, inPts };
+
+    return runOCVnGAPIOptFlowLK<cv::GMat>(input,
+                                          static_cast<size_t>(testInst.in_mat1.cols),
+                                          static_cast<size_t>(testInst.in_mat1.rows),
+                                          params,
+                                          ocvOut,
+                                          gapiOut);
+}
+
+inline cv::GComputation runOCVnGAPIOptFlowLKForPyr(TestFunctional& testInst,
+                                                   OptFlowLKTestInput<std::vector<cv::Mat>>& in,
+                                                   const OptFlowLKTestParams& params,
+                                                   bool withDeriv,
+                                                   OptFlowLKTestOutput& ocvOut,
+                                                   OptFlowLKTestOutput& gapiOut)
+{
+    testInst.initMatsFromImages(params.channels,
+                                params.fileNamePattern,
+                                params.format);
+
+    cv::Size winSize(params.winSize, params.winSize);
+
+    OptFlowLKTestParams updatedParams(params);
+    updatedParams.maxLevel = cv::buildOpticalFlowPyramid(testInst.in_mat1, in.prevData,
+                                                         winSize, params.maxLevel, withDeriv);
+    updatedParams.maxLevel = cv::buildOpticalFlowPyramid(testInst.in_mat2, in.nextData,
+                                                         winSize, params.maxLevel, withDeriv);
+
+
+    return runOCVnGAPIOptFlowLK<cv::GArray<cv::GMat>>(in,
+                                                      static_cast<size_t>(testInst.in_mat1.cols),
+                                                      static_cast<size_t>(testInst.in_mat1.rows),
```

Comment:
either use checked_cast from ADE or make parameters integers.

Reply:
Is it good to link  perf_tests  with `ade` to use this?

Reply:
Well, no, then better fix the types

Reply:
as I can see it's impossible to fix the types totally: on the one hand, the function got `int`'s - from `Mat.cols` and `Mat.rows` members; on the other - `std::vector.resize()` takes `size_t`...
not to resize vector before filling in cycle is bad, I think
can we leave the only `static_cast` (on line 48 this file) and check Assert before it?